apollo-portal/src/main/java/com/ctrip/framework/apollo/openapi/service/AuthorizationService.java

```diff
@@ -0,0 +1,70 @@
+package com.ctrip.framework.apollo.openapi.service;
+
+import com.ctrip.framework.apollo.common.entity.App;
+import com.ctrip.framework.apollo.openapi.dto.OpenAppDTO;
+import com.ctrip.framework.apollo.openapi.entity.ConsumerRole;
+import com.ctrip.framework.apollo.openapi.repository.ConsumerRoleRepository;
+import com.ctrip.framework.apollo.openapi.util.OpenApiBeanUtils;
+import com.ctrip.framework.apollo.portal.entity.po.Role;
+import com.ctrip.framework.apollo.portal.repository.RoleRepository;
+import com.ctrip.framework.apollo.portal.service.AppService;
+import com.ctrip.framework.apollo.portal.util.RoleUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * @author wxq
+ */
+@Service
+public class AuthorizationService {
+
+  private final ConsumerService consumerService;
+
+  private final AppService appService;
+
+  @Autowired
+  private ConsumerRoleRepository consumerRoleRepository;
+
+  @Autowired
+  private RoleRepository roleRepository;
+
+  public AuthorizationService(
+      ConsumerService consumerService,
+      AppService appService) {
+    this.consumerService = consumerService;
+    this.appService = appService;
+  }
+
+  private Set<String> getAppIdsByRoleIds(List<Long> roleIds) {
+    Iterable<Role> roleIterable = this.roleRepository.findAllById(roleIds);
```

Comment:
Should we use `com.ctrip.framework.apollo.portal.service.RolePermissionService` instread of using RoleRepository and ConsumerRoleRepository  directly?

Reply:
I'm thinking may we don't need to add this new `AuthorizationService` but add a `findConsumerRoles` method in `ConsumerRolePermissionService` and put other logics in AppController, just like com.ctrip.framework.apollo.portal.controller.AppController#findAppsByOwner

https://github.com/ctripcorp/apollo/blob/de05540e6c9a18cdf370ae95ca5e544d7b8d04bf/apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/controller/AppController.java#L91-L106